### DISCUSSION TITLE
Polar contour plot

### DISCUSSION THREAD

**Discussion by psalvaggio** [alandefreitas/matplotplusplus#117] [Q&A]
Hi, I am trying to plot a bidrectional reflectance distribution function (BRDF) with this library. Traditionally, this is done as a filled contour plot in polar space. I have done this in the past with `matplotlib`, but I didn't see it in the gallery of examples. Is it possible in matplotplusplus? Thanks!

**Comment by alandefreitas** [comment]
The `polar` function returns a `line_handle` (a shared_ptr to a `line`)

https://github.com/alandefreitas/matplotplusplus/blob/159a438a25760fd6c3b77df202ebcb838164bb2c/source/matplot/core/axes_type.h#L1309

A `line` plot has a `fill` property 

https://github.com/alandefreitas/matplotplusplus/blob/159a438a25760fd6c3b77df202ebcb838164bb2c/source/matplot/axes_objects/line.h#L77

that might give you what want if you set it to `true`. Something like:

```cpp
auto h = ax->polar(...); // or matplot::polar(...)
// and then
h->fill(true);
```

But I haven't tested that yet but it should work, because polar histograms also use filled lines:

![](https://alandefreitas.github.io/matplotplusplus/examples/polar_plots/polarhistogram/polarhistogram_1.svg)

https://github.com/alandefreitas/matplotplusplus/blob/master/examples/polar_plots/polarhistogram/polarhistogram_1.cpp

**Reply by psalvaggio** [reply]
so looking at `polarplot`, it takes a 1D array of theta/rho values. I have more of a 2D dataset, with a set of zenith/azimuth (which map to r/theta), with a function value at each. The plot I'm trying to make looks like:
![image](https://user-images.githubusercontent.com/13050294/103036291-6aaa5e00-4537-11eb-9d4a-014a049b7221.png)
I think what I'm after is just a way to make `contourf` draw with polar axes instead. Is that possible?

**Comment by alandefreitas** [comment]
Mmmmm.... You might be able to achieve that with contourc then. Same as above (get handle, fill(true), etc...), but give the polar plot polygons for the contour lines instead. You can calculate the contour lines with contourc. (Still have never tested any of that though)

**Reply by psalvaggio** [reply]
OK, thanks. I will give it a try!
